source/fab/cli.py

```diff
@@ -0,0 +1,83 @@
+# ##############################################################################
+#  (c) Crown copyright Met Office. All rights reserved.
+#  For further details please refer to the file COPYRIGHT
+#  which you should have received as part of this distribution
+# ##############################################################################
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict, Optional
+
+import fab
+from fab.artefacts import CollectionGetter
+from fab.build_config import BuildConfig
+from fab.constants import PRAGMAD_C
+from fab.steps.analyse import Analyse
+from fab.steps.c_pragma_injector import CPragmaInjector
+from fab.steps.compile_c import CompileC
+from fab.steps.compile_fortran import CompileFortran, get_fortran_compiler, get_fortran_preprocessor
+from fab.steps.find_source_files import FindSourceFiles
+from fab.steps.grab.folder import GrabFolder
+from fab.steps.link import LinkExe
+from fab.steps.preprocess import c_preprocessor, fortran_preprocessor
+from fab.steps.root_inc_files import RootIncFiles
+
+
+def _generic_build_config(folder: Path, kwargs: Optional[Dict] = None) -> BuildConfig:
+    folder = folder.resolve()
+    kwargs = kwargs or {}
+
+    # Within the fab workspace, we'll create a project workspace.
+    # Ideally we'd just use folder.name, but to avoid clashes, we'll use the full absolute path.
+    label = '/'.join(folder.parts[1:])
```

Comment:
Is there  a way to do this using only Path objects so as to avoid string handling? Is it important to do that if the resulting label is needed as a string anyway?

Reply:
We _could_ change `label` to be a path, as it's mainly used to create a subfolder in the fab workspace. I'm not sure we should bother doing that, so yes, it's probably not important, as you say.